modules/core/src/test/scala/com/azavea/stac4s/CatalogSpec.scala

```diff
@@ -0,0 +1,436 @@
+            "landsat:wrs_row"                      -> 25.asJson,
+            "landsat:earth_sun_distance"           -> 1.0141560.asJson,
+            "landsat:ground_control_points_verify" -> 114.asJson,
+            "landsat:geometric_rmse_model"         -> 7.562.asJson,
+            "landsat:image_quality_tirs"           -> 9.asJson,
+            "landsat:ground_control_points_model"  -> 313.asJson,
+            "landsat:geometric_rmse_model_x"       -> 5.96.asJson,
+            "landsat:geometric_rmse_model_y"       -> 4.654.asJson,
+            "landsat:geometric_rmse_verify"        -> 5.364.asJson,
+            "landsat:image_quality_oli"            -> 9.asJson
+          )
+        ),
+        links = List(
+          StacLink(
+            href = "../../catalog.json",
+            rel = StacLinkType.StacRoot,
+            _type = None,
+            title = None,
+            labelExtAssets = Nil
+          ),
+          StacLink(
+            href = "../catalog.json",
+            rel = StacLinkType.Parent,
+            _type = None,
+            title = None,
+            labelExtAssets = Nil
+          ),
+          StacLink(
+            href = "LC81530252014153LGN00.json",
+            rel = StacLinkType.Self,
+            _type = None,
+            title = None,
+            labelExtAssets = Nil
+          ),
+          //  { "rel":"alternate", "href": "https://landsatonaws.com/L8/153/025/LC81530252014153LGN00", "type": "text/html"},
+          StacLink(
+            href = "https://landsatonaws.com/L8/153/025/LC81530252014153LGN0",
+            rel = StacLinkType.Alternate,
+            _type = Some(`text/html`),
+            title = None,
+            labelExtAssets = Nil
+          )
+        ),
+        assets = Map(
+          "thumbnail" -> StacItemAsset(
+            href =
+              "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_thumb_large.jpg",
+            title = Some("Thumbnail"),
+            description = Some("A medium sized thumbnail"),
+            roles = List(StacAssetRole.Thumbnail),
+            _type = Some(`image/jpeg`)
+          ),
+          "metadata" -> StacItemAsset(
+            href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_MTL.txt",
+            title = Some("Original Metadata"),
+            description = Some("The original MTL metadata file provided for each Landsat scene"),
+            roles = List(StacAssetRole.Metadata),
+            _type = Some(VendorMediaType("mtl"))
+          ),
+          "B1" -> StacItemAsset(
+            href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_B1.TIF",
+            title = Some("Coastal Band (B1)"),
+            description = Some("Coastal Band Top Of the Atmosphere"),
+            roles = Nil,
+            _type = Some(`image/tiff`)
+            // "eo:bands": [0],
```

Comment:
What options do you see for adding support for these types of properties?

If we go the route like we did for `label:assets` we have to make this optional I think on `StacItemAsset`. I think alternative, but could be kind of laborious, is to have an ADT maybe for things like this. It could get out of hand pretty quickly, but it would be something like:

```scala
sealed trait StacItemAsset {
... <min required attributes >
}
case class EOAsset(<eo attributes>) extends StacItemAsset {
... 
}
case class LabelAsset(<label attribute>) extends StacItemAsset {
... 
}
case class EOLabelAsset(<both attributes>) extends StacItemAsset {
}
```

The same logic would have to apply to other types that have custom attributes based on the extension as well. `properties` will always be kind of a mess since it will always be possible to put anything in there.

Reply:
@notthatbreezy what if you need Assets that have some mixin fields from different extensions?

I know that this is a bad example, but i.e. a regular usecase is `view` + `eo` + `landsat`. What if some of them add multiple fields into Stac classes?

Reply:
Yeah - I think that's why I think it would be labor intensive - for every combination of every extension we would need to be exhaustive. So if there are custom attributes on `StacItem` for an extension, we'd need to have `EOStacItem`, `LabelStacItem`, `EOLabelStacItem`, `ViewStacItem`, `EOLabelViewStacItem`, `LabelViewStacItem` etc.

I don't think it's a good idea necessarily, but just trying to brainstorm.

Reply:
I think another option is to carry around `extensionProperties: JsonObject` and do the lazy decoding / validation that we talked about with items. It would make the decoder a bit weird I think, where we'd have to pop off the truly required properties and then somehow parse "the rest", but it would give us flexibility to keep everything, even invalid items, and push the choice of whether it matters if assets are valid eo assets (or label assets or ...) to users.